LogicalTypes.md

```diff
@@ -366,3 +366,27 @@ optional group my_map (MAP_KEY_VALUE) {
 }
 ```
 
+### Unions
+
+A Union type annotates data stored as a Group.
+It describes the different possible types under the same field name.
+The names of the fields in the annotated Group are not important in such a case, but
+as a convention the type names are used.
+All fields of the Group must be optional and exactly one is defined for each instance of the group.
+If more than one is defined the behavior is undefined and may changed depending on the projection applied.
+A Union can not contain null but can be null itself if in an optional field.
+
+// Union<String, Integer, Boolean> (nullable union of either String, Integer or Boolean)
```

Comment:
Unlike Avro, in Parquet Null is not a type. Just like Avro Records do not have optional fields.
So I think that this spec should not treat Null as a type.
It is clearer to have one-to-one mapping between the types here and the fields in the Group since this describes how it is stored.

If we want to clarify this, I can add the following:
Mapping to Avro types:
 - an Avro Union that contains Null and at least 2 other types will map to an optional Parquet Union (of the remaining types).
 - an Avro Union that does not contain null will map to a required Parquet Union.